client/blocks/signup-form/passwordless.jsx

```diff
@@ -0,0 +1,260 @@
+				email: typeof this.state.email === 'string' ? this.state.email.trim() : '',
+				is_passwordless: true,
+				signup_flow_name: this.props.flowName,
+				validate: false,
+				locale: getLocaleSlug(),
+				client_id: config( 'wpcom_signup_id' ),
+				client_secret: config( 'wpcom_signup_key' ),
+			} );
+			this.createAccountCallback( null, response );
+		} catch ( err ) {
+			this.createAccountCallback( err );
+		}
+	};
+
+	createAccountCallback = ( error, response ) => {
+		if ( error ) {
+			const errorMessage = this.getErrorMessage( error );
+			this.setState( {
+				errorMessages: [ errorMessage ],
+				isSubmitting: false,
+			} );
+			this.submitTracksEvent( false, { action_message: error.message } );
+			return;
+		}
+
+		this.setState( {
+			errorMessages: null,
+			isSubmitting: false,
+		} );
+
+		const username =
+			( response && response.signup_sandbox_username ) || ( response && response.username );
+
+		const userId =
+			( response && response.signup_sandbox_user_id ) || ( response && response.user_id );
+
+		const userData = {
+			ID: userId,
+			username: username,
+			email: this.state.email,
+		};
+
+		const marketing_price_group = response?.marketing_price_group ?? '';
+
+		recordRegistration( {
+			userData,
+			flow: this.props.flowName,
+			type: 'passwordless',
+		} );
+
+		this.submitStep( {
+			username,
+			marketing_price_group,
+			bearer_token: response.bearer_token,
+		} );
+	};
+
+	getErrorMessage( errorObj = { error: null, message: null } ) {
+		const { translate } = this.props;
+
+		switch ( errorObj.error ) {
+			case 'already_taken':
+			case 'already_active':
+			case 'email_exists':
+				return (
+					<>
+						{ translate( 'An account with this email address already exists.' ) }
+						&nbsp;
+						{ translate( '{{a}}Log in now{{/a}} to finish signing up.', {
+							components: {
+								a: (
+									<a
+										href={ `${ this.props.logInUrl }&email_address=${ encodeURIComponent(
+											this.state.email
+										) }` }
+									/>
+								),
+							},
+						} ) }
+					</>
+				);
+			default:
+				return translate(
+					'Sorry, something went wrong when trying to create your account. Please try again.'
+				);
+		}
+	}
+
+	submitStep = ( data ) => {
+		const { flowName, stepName, goToNextStep, submitCreateAccountStep } = this.props;
+		submitCreateAccountStep(
+			{
+				flowName,
+				stepName,
+				// We use this flag in the flow controller to communicate to the flow controller that we're
+				// dealing with passwordless signup, and therefore don't need to call the apiFunction
+				// normally associated with the user step.
+				isPasswordlessSignupForm: true,
+			},
+			data
+		);
+		this.submitTracksEvent( true, { action_message: 'Successful login', username: data.username } );
+		goToNextStep();
+	};
+
+	onInputChange = ( { target: { value } } ) =>
+		this.setState( {
+			email: value,
+			errorMessages: null,
+			isEmailAddressValid: emailValidator.validate( value ),
```

Comment:
It seems like the `isEmailAddressValid` component state property is not used and can be removed.